File: Leetcode/Problems/HouseRobber/house_robber.cpp
#include <iostream>
using namespace std;
int rob(vector<int> nums)
{
    vector<int> maximums(nums.size(), 0);
    if (nums.size() == 1)
        return nums[0];
    maximums[0] = nums[0];
    maximums[1] = max(nums[0], nums[1]);
    for (int i = 2; i < nums.size(); i++)
    {
        maximums[i] = max(maximums[i - 1], maximums[i - 2] + nums[i]);
    }
    return maximums[nums.size() - 1];
}
int main()
{
    vector<int> nums = {2, 1, 1, 2};
    cout << rob(nums) << endl;
}

File: Leetcode/Problems/Solution/HouseRobber.java
public class HouseRobber {

  public static int rob(int[] nums) {
    int[] maxAmount = new int[nums.length];
    maxAmount[0] = nums[0];
    for (int i = 1; i < nums.length; i++) {
      if (i == 1) {
        maxAmount[i] = Math.max(maxAmount[i - 1], nums[i]);
      } else {

        maxAmount[i] = Math.max(maxAmount[i - 1], maxAmount[i - 2] + nums[i]);
      }

    }
    return maxAmount[nums.length - 1];
  }

  public static void main(String[] args) {
    int[] nums = {0};
    System.out.println(rob(nums));
  }
}
